### PR TITLE
Add mock-block-dock to all Blocks

### PR DESCRIPTION
**Pull request by akash-joshi**
## 🌟 What is the purpose of this PR?

Hash has recently published an npm package for mocking an embedding application called [mock-block-dock](https://npmjs.com/package/mock-block-dock), which acts as a testing ground for blocks. This package is also used to wrap the demo blocks in the BlockProtocol website. However, it wasn't used extensively in the local dev setup of the blocks present in the Hash repo.

Thus, this PR adds mock-block-dock as a wrapper and dependency to all existing blocks, allowing us to reuse test functionality across blocks and also catch any errors early.

## 🔍 What does this change?

- Adds `mock-block-dock` as a dev dependency to all blocks.
- Wraps blocks in its wrapper and removes any local block-protocol function implementations.
- Refactors `.js` files to `.tsx` where needed for better code formatting and linting.

## 🛡 What tests cover this?

- Existing linting and TS checks should cover any compiler-side errors.

## ❓ How to test this?

1. Checkout the branch.
2. Run `yarn` on the root directory to install dependencies.
3. Visit each block's directory and run `yarn dev`.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201955153125185

### CODE REVIEW

File: packages/blocks/person/src/index.css [hunk 1]
```diff
@@ -54,6 +54,10 @@
   align-items: center;
 }
 
+.person-link svg {
+  transform: translateY(2px);
```
Comment (akash-joshi) [inline packages/blocks/person/src/index.css:58]: Additional change: [Fix link icon offset](https://app.asana.com/0/1200211978612931/1201961850830767/f)

chrome_NqrEdeV9aB

**Review by CiaranMn — CHANGES_REQUESTED** [review]
Thanks Akash. `MockBlockDock` provides `createLinks`, `uploadFile` etc, so you can remove those from the dev renderers too. 

The only exception is `getEmbedCode`.

File: packages/blocks/code/src/webpack-dev-server.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/**
+ * This is the entry point for developing and debugging.
+ * This file is not bundled with the library during the build process.
+ */
+import React from "react";
+import ReactDOM from "react-dom";
+import { tw } from "twind";
+import { MockBlockDock } from "mock-block-dock";
+
+// eslint-disable-next-line import/extensions
```
Comment (CiaranMn) [inline packages/blocks/code/src/webpack-dev-server.tsx:10]: This is no longer necessary

**Review by CiaranMn — DISMISSED** [review]

File: packages/blocks/code/src/webpack-dev-server.tsx (hunk 2 above)
Comment (akash-joshi) [reply packages/blocks/code/src/webpack-dev-server.tsx]: Done: https://github.com/hashintel/hash/pull/415/commits/2409f56abeac464729193192ad076d44493ed83d

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**